tests/test_dojo.py

```diff
@@ -76,21 +111,121 @@ def test_print_room(self):
+    def test_print_room_not_added(self):
+        '''test that members of a room are created'''
+        self.dojo.create_room('', 'office')
+        self.dojo.create_room('', 'livingspace')
+        self.dojo.add_person('sonia', 'fellow')
+        self.dojo.add_person('kobby', 'staff')
+        self.dojo.add_person('bett', 'fellow')
+        self.dojo.print_room('blue')
+
+    def test_invalid_print_room(self):
+        '''test invalid room '''
         self.dojo.create_room('blue', 'office')
         self.dojo.create_room('brown', 'livingspace')
+        self.dojo.add_person('sonia', 'fellow' 'Y')
         self.dojo.add_person('kobby', 'staff')
         self.dojo.add_person('bett', 'fellow')
-        self.dojo.add_person('khalid', 'fellow', 'Y')
-        self.dojo.print_allocations('allocations.txt')
+        self.dojo.print_room('green')
 
+    def test_print_allocations_filename(self):
+        '''test that allocated people are printed to a file'''
+        self.dojo.create_room('BLUE', 'office')
+        self.dojo.add_person('kobby', 'staff')
+        self.dojo.print_allocations('allocated')
+        self.assertTrue(os.path.isfile('allocated.txt')) 
+        file = open('allocated.txt', 'r')
+        output = file.readlines()
+        first_line = output [0]
+        second_line = output [2]
+        self.assertEquals(first_line, 'BLUE' + "\n")
+        self.assertEquals(second_line, 'Kobby' + "\n")
 
     def test_print_unallocated_filename(self):
         '''test that unallocated people are printed to a file'''
-        self.dojo.create_room('blue', 'office')
-        self.dojo.create_room('brown', 'livingspace')
-        self.dojo.add_person('kobby', 'staff')
-        self.dojo.add_person('bett', 'fellow')
-        self.dojo.add_person('khalid', 'fellow', 'Y')
-        self.dojo.print_unallocated('unallocated.txt')                
\ No newline at end of file
+        self.dojo.add_person('kobby', 'fellow','Y')
+        self.dojo.print_unallocated('unallocated') 
+        self.assertTrue(os.path.isfile('unallocated.txt'))
+        unallocated_file = open('unallocated.txt', 'r')
+        text = unallocated_file.readlines()
+        first_line = text [0]
+        second_line = text [1]
+        third_line = text[3]
+        self.assertEquals(first_line, "=" * 20 + "\n")
+        self.assertEquals(second_line, 'Unallocated People' + "\n")
+        self.assertEquals(third_line, "Kobby" + '\n') 
+
+    #    tests for reallocate person starts here --->
+    def test_reallocate_person(self):
+        '''test that a person has been reallocated to a different room'''
+        self.dojo.create_room('hogwarts', 'office')
+        self.dojo.add_person('john', 'staff')
+        staff = self.dojo.staff[0]
+        office_name = staff.office.name
+        self.assertEqual(office_name, 'hogwarts')
+        self.dojo.create_room('swift', 'office')
+        self.dojo.reallocate_person('joshua', 'swift')
```

Comment:
what uniquely identifies the person to be reallocated, in the case where two people have the same name